src/main/java/com/iexec/sms/secret/teetaskruntime/AppRuntimeSecretController.java

```diff
@@ -0,0 +1,315 @@
+                OnChainObjectType.APPLICATION,
+                appAddress,
+                SecretOwnerRole.APPLICATION_DEVELOPER,
+                null,
+                secretIndex
+        );
+        if (isSecretPresent) {
+            log.debug("App developer secret found [appAddress: {}, secretIndex: {}]", appAddress, secretIndex);
+            return ResponseEntity.noContent().build();
+        }
+
+        log.debug("App developer secret not found [appAddress: {}, secretIndex: {}]", appAddress, secretIndex);
+        return ResponseEntity
+                .status(HttpStatus.NOT_FOUND)
+                .body(createErrorPayload("Secret not found"));
+    }
+
+    @PostMapping("/apps/{appAddress}/requesters/secrets")
+    public ResponseEntity<Map<String, String>> setMaxRequesterSecretCountForApp(
+            @RequestHeader("Authorization") String authorization,
+            @PathVariable String appAddress,
+            @RequestBody int secretCount) {
+        String challenge = authorizationService
+                .getChallengeForSetRequesterAppRuntimeSecretCount(
+                        appAddress,
+                        secretCount
+                );
+
+        if (!authorizationService.isSignedByOwner(challenge, authorization, appAddress)) {
+            log.error("Unauthorized to setRequesterSecretCountForApp" +
+                            " [appAddress: {}, expectedChallenge: {}]",
+                    appAddress, challenge);
+            return ResponseEntity
+                    .status(HttpStatus.UNAUTHORIZED)
+                    .body(invalidAuthorizationPayload);
+        }
+
+        final boolean isCountAlreadyPresent = teeTaskRuntimeSecretCountService
+                .isMaxAppRuntimeSecretCountPresent(
+                        appAddress,
+                        SecretOwnerRole.REQUESTER
+                );
+        if (isCountAlreadyPresent) {
+            log.debug("Can't add app requester app secret count as it already exist"
+            + " [appAddress:{}]", appAddress);
+            return ResponseEntity
+                    .status(HttpStatus.CONFLICT)
+                    .body(createErrorPayload("Secret count already exists"));
+        }
+
+        final boolean hasBeenInserted = teeTaskRuntimeSecretCountService.setMaxAppRuntimeSecretCount(
+                appAddress,
+                SecretOwnerRole.REQUESTER,
+                secretCount
+        );
+
+        if (!hasBeenInserted) {
+            return ResponseEntity
+                    .badRequest()
+                    .body(createErrorPayload(
+                            "Secret count should be positive. " +
+                            "Can't accept value " + secretCount
+                    ));
+        }
+
+        return ResponseEntity.noContent().build();
+    }
+    // endregion
+
+    // region App requester endpoint
+    @PostMapping("/requesters/{requesterAddress}/apps/{appAddress}/secrets/{secretIndex}")
```

Comment:
It seems somehow strange that the route `/requesters` is inside `App...Controller`!

Reply:
If we keep it here, we can probably invert the route hierarchy like this:
```/apps/{appAddress}/requesters/{requesterAddress}/secrets/{secretIndex}```.

Reply:
2 points that were discussed with @jeremyjams:
- `App...Controller` relates to all app runtime/compute runtime secrets. Requester secrets are part of it, so it seems OK to let these routes there :)
- `/requesters/{requesterAddress}/apps/{appAddress}/secrets/{secretIndex}` creates something like a folder, identified by `/requesters/{requesterAddress}/` within which a requester can find everything that relates to them. Reversing the route as you suggest would make that unclear for the requester. Besides, we usually don't mind to find all requester secrets for a given app - that's what would suggest the route you propose.

Reply:
* We agree on the 1st point if you mean `AppCompute` secrets (just to be precise).
* I understand that `/requesters/{requesterAddress}/` creates a folder of secrets for a requester, but those secrets, if they are managed inside `App...Controller`, are only usable in the context `app`. If we want an endpoint that controls all secrets of a requester than it's more intuitive to use a `RequesterSecretController`. Otherwise, were would we put (for example) requester's `postCompute` secrets?

Reply:
Regarding your last question: IMHO, it should go in a `PostComputeSecretController` if we keep the current idea.
If we decide to migrate `/requesters/{requesterAddress}/` routes to a new `RequesterSecretController`, then we could rename the current `AppComputeSecretController` to `AppDeveloperComputeSecretController`... I really don't know which is best.

Reply:
> Regarding your last question: IMHO, it should go in a PostComputeSecretController if we keep the current idea.

Exactly, and in that case it wouldn't start with `/requesters/{id}/....`. Same goes for `AppCompute`, it's more coherent to make the root route `/app/...`.

Reply:
Well... I disagree :smile: 
If we keep the current idea, it would be something like `/requesters/{requesterAddress}/post-compute` in `PostComputeSecretController`.
I think it really depends on how you see the relation between the secrets, the runtime/compute context and the secret owner.

----
As a first thought, I had the same idea as you: having a dedicated `RequesterSecretController`. But @jeremyjams thought it was a good solution to put `/requesters/{requesterAddress}/app` routes into the `AppComputeSecretController` and convinced me to do so. [See there](https://github.com/iExecBlockchainComputing/iexec-sms/pull/69#discussion_r758126265). Maybe he has other inputs.

Reply:
I find it strange that the endpoint `/requesters/...` be split between two controllers.